src/bidiMapper/domains/storage/StorageProcessor.ts

```diff
@@ -42,6 +42,46 @@ export class StorageProcessor {
     this.#logger = logger;
   }
 
+  async deleteCookies(
+    params: Storage.DeleteCookiesParameters
+  ): Promise<Storage.DeleteCookiesResult> {
+    const partitionKey = this.#expandStoragePartitionSpec(params.partition);
+
+    const cdpResponse = await this.#browserCdpClient.sendCommand(
+      'Storage.getCookies',
+      {
+        browserContextId: partitionKey.userContext,
+      }
+    );
+
+    const cdpCookiesToDelete = cdpResponse.cookies
+      .filter(
+        // CDP's partition key is the source origin. If the request specifies the
+        // `sourceOrigin` partition key, only cookies with the requested source origin
+        // are returned.
+        (c) =>
+          partitionKey.sourceOrigin === undefined ||
+          c.partitionKey === partitionKey.sourceOrigin
+      )
+      .filter((cdpCookie) => {
+        const bidiCookie = cdpToBiDiCookie(cdpCookie);
+        return this.#matchCookie(bidiCookie, params.filter);
+      })
+      .map((cookie) => ({
+        ...cookie,
+        // Set expiry to pass date to delete the cookie.
+        expires: 1,
```

Comment:
I believe it's a valid passed date.

Reply:
Another approach would be to remove all the cookies via `Storage.clearCookies` and set those didn't have to be deleted. But this would create a race condition with a concurrent read.